### ISSUE TITLE
Use of deprecated `std::str::trim_left`

### ISSUE THREAD

**Issue by HeroicKatora** [PistonDevelopers/image#852]
# Description

This is an early but easy warning. As the title suggest suggests, the `pnm` decoder makes use of `str::trim_left` which was deprecated and replaced by `str::trim_start` starting in Rust `1.33`. The reason for that change is no other than clarity of function name, as trim works in byte-order not the unicode text direction as the use of 'left' or 'right' could wrongly imply.

# Warning

```
warning: use of deprecated item 'core::str::<impl str>::trim_left': superseded by `trim_start`
   --> src/pnm/decoder.rs:293:43
    |
293 |             let (identifier, rest) = line.trim_left()
    |                                           ^^^^^^^^^
    |
    = note: #[warn(deprecated)] on by default
```

# Fix

The most progressive way would be to replace the functions with their new alias `str::trim_start`. But as this was introduced only in `1.30.0` we would need to stay backwards compatible–we aim to support down to `1.24.1` I believe [in favour of Ubuntu](https://packages.ubuntu.com/search?keywords=rustc&searchon=names&suite=all&section=all) and similar slow moving distros. 

Since [`cfg`-like selection based on version or accessible features](https://github.com/rust-lang/rfcs/pull/2523) will unlikely be available before, marking *this specific call* with `#[allow(deprecated)]` is better for the time being.

**Comment by lazypassion** [comment]
I understand wanting backwards compatibility, but it seems like rust is still very close to bleeding edge and the library ecosystem isn't mature yet, including this library. So i'm just curious why having backwards compatibility this "early" would matter? Just wondering.

**Comment by HeroicKatora** [comment]
Fixed in #857 , thanks
